src/scanpy/preprocessing/_qc.py

```diff
@@ -393,6 +413,34 @@ def top_segment_proportions(
     """
     Calculates total percentage of counts in top ns genes.
 
+    Parameters
+    ----------
+    mtx
+        Matrix, where each row is a sample, each column a feature.
+    ns
+        Positions to calculate cumulative proportion at. Values are considered
+        1-indexed, e.g. `ns=[50]` will calculate cumulative proportion up to
+        the 50th most expressed gene.
+    """
+    # Pretty much just does dispatch
+    if not (max(ns) <= mtx.shape[1] and min(ns) > 0):
+        raise IndexError("Positions outside range of features.")
+    if isinstance(mtx, DaskArray):
+        if not isinstance(mtx._meta, csr_matrix | np.ndarray):
+            msg = f"DaskArray must have csr matrix or ndarray meta, got {mtx._meta}."
+            raise ValueError(msg)
+        return mtx.map_blocks(
+            lambda x: top_segment_proportions_in_memory(mtx=x, ns=ns), meta=np.array([])
+        ).compute()
+    return top_segment_proportions_in_memory(mtx, ns)
+
+
+def top_segment_proportions_in_memory(
+    mtx: np.ndarray | spmatrix, ns: Collection[int]
+) -> np.ndarray:
+    """
+    Calculates total percentage of counts in top ns genes.
```

Comment:
please delete the docstring here, this is just a helper function to dispatch to, people should go through `top_segment_proportions`, right?

Reply:
why should we delete the docstring, seems better than not having one?

Reply:
it’s duplicated, so when someone modifies it, there’s a >0 chance that they only modify one version and not both, which is worse than having to search for the docstring for a few seconds.

if you want, you can rename these internal helper functions to `_leading_underscore` names, then it’s even more likely that future us will find the entrypoint `top_proportions` and its docstring.

Reply:
great point, refactored for singledispatch